### PR TITLE
DOC: Add ITK scientific ecosystem section to documentation

### PR DESCRIPTION
**Pull request by jhlegarreta**
Add ITK scientific ecosystem section to documentation:
- Include other tools developed around ITK that allow to leverage its use beyond its natural boundaries.
- Transition the wiki command line tools page: https://itk.org/Wiki/ITK/CommandLineTools
- Mention `ITKApps` as a set of legacy tools that leveraged ITK's capabilities without completely transitioning the corresponding https://itk.org/Wiki/ITK/Insight_Applications wiki page, as the instructions refer to old releases and tools.

## PR Checklist
- [X] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [X] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)

### CODE REVIEW

**Comment by jhlegarreta** [comment]
Notes:
- Feel free to propose changes to the name, contents, etc.
- Re https://itk-viewer.on.fleek.co/: Not sure about the status of this. Points to IPFS data, not a viewer.
- Re https://itk.org/Wiki/ITK/Insight_Applications: Don't think transferring this is useful: the application repository is archived, and maybe some effort would be necessary to make things work on modern systems. The Wiki page mentions ITK versions as old as 1.6, Qt 2.0, and VTK 4.2

Slightly related:
- I do not think the links/contents in https://itk.org/Wiki/ITK/Developer_Software are relevant any more:
  - Pretty debugging: https://itk.org/Wiki/ITK/GDBPretty : not sure how this was/can be useful.
  - Debug Visualizers for Visual Studio: https://itk.org/Wiki/ITK/Debug_Visualizers_for_Visual_Studio : refers to VS 2018, 2010, 2012: deprecated.
- **But** CMake: points to https://gitlab.kitware.com/cmake/community/-/wikis/Home **could be linked from somewhere else** as it looks maintained/relevant

**Review by thewtex — APPROVED** [review]
:heart:

File: Documentation/docs/scientific_ecosystem/itk_ecosystem.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+- [itk-wasm](https://wasm.itk.org/en/latest/) combines ITK and
+  [WebAssembly](https://webassembly.org/) to enable high-performance
+  spatial analysis in a web browser or system-level environments and
+  reproducible execution across programming languages and hardware
+  architectures.
+- [itkwidgets](https://itkwidgets.readthedocs.io/en/latest/) is an
+  elegant Python interface for visualization on the web platform to
+  interactively generate insights into multidimensional images, point
+  sets, and geometry.
+
```
[the remark this answers is not in the document]
Comment (jhlegarreta) [reply Documentation/docs/scientific_ecosystem/itk_ecosystem.md]: Added in f6008cb.

**Review by hjmjohnson — APPROVED** [review]

File: Documentation/docs/scientific_ecosystem/itk_ecosystem.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  reproducible execution across programming languages and hardware
+  architectures.
+- [itkwidgets](https://itkwidgets.readthedocs.io/en/latest/) is an
+  elegant Python interface for visualization on the web platform to
+  interactively generate insights into multidimensional images, point
+  sets, and geometry.
+- [scikit-build](https://scikit-build.org) is a Python build system for
+  CPython C/C++/Fortran/Cython extensions using [CMake](https://cmake.org).
+
+## Legacy tools
```
Comment (jhlegarreta) [inline Documentation/docs/scientific_ecosystem/itk_ecosystem.md:23]: @thewtex Should we remove this as per https://github.com/InsightSoftwareConsortium/ITK/pull/4313#discussion_r1390272297?

**Pull request merged**